lua/lazyvim/plugins/extras/lang/fsharp.lua

```diff
@@ -0,0 +1,63 @@
+-- requires dotnet sdk  to be installed
+return {
+  recommended = function()
+    return LazyVim.extras.wants({
+      ft = { "fs", "fsx", "fsi" },
+      root = { "*.fsproj" },
+    })
+  end,
+  {
+    "nvim-treesitter/nvim-treesitter",
+    config = function()
+      local parser_config = require("nvim-treesitter.parsers").get_parser_configs()
+      parser_config.fsharp = {
+        install_info = {
+          url = "https://github.com/ionide/tree-sitter-fsharp",
+          branch = "main",
+          files = { "src/scanner.c", "src/parser.c" },
+        },
+        filetype = "fsharp",
+      }
+    end,
+    opts = { ensure_installed = { "fsharp" } },
+  },
+  {
+    "ionide/Ionide-vim",
```

Comment:
Yeah it provides the functionality which omnisharp-extended-lsp provides, like full project scanning and completions, which wouldn't work properly without this plugin, because the fsharp_language_server stated in server_configurations in nvim lspconfig is no longer maintained, and is stopped at .net 6 i guess, and current lts is .net 8